writer_test.go

```diff
@@ -435,3 +435,82 @@ func TestWriterWithNils(t *testing.T) {
 		t.Errorf("Test failed, expected %v, got %v", numValues, row)
 	}
 }
+
+func TestBooleanWriterAlternating(t *testing.T) {
+	// Test was added to address issue: https://github.com/scritchley/orc/issues/61
+	// Involves writing alternative true, nil, false values to a boolean column.
+
+	f, err := ioutil.TempFile("", "testorc")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	filename := f.Name()
+	defer os.Remove(filename)
+	defer f.Close()
+
+	schema, err := ParseSchema("struct<boolean1:boolean>")
+	if err != nil {
+		t.Fatal(err)
+	}
+	w, err := NewWriter(f, SetSchema(schema))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	length := 20000
+	expected := make([]interface{}, length)
+
+	for i := 0; i < length; i++ {
+		if i%2 == 0 {
+			err = w.Write(true)
+			expected[i] = true
+
+		} else if i%3 == 0 {
+			err = w.Write(nil)
+			expected[i] = nil
+		} else {
+			err = w.Write(true)
+			expected[i] = true
+		}
```

Comment:
Do you want to make one of these boolean values false just so each case is covered?

Reply:
ah good catch, I changed this briefly whilst debugging and forgot to revert it.